### ISSUE TITLE
Deleting more than 5,000 contact entries crashes SimpleContacts

### ISSUE THREAD

**Issue by eklausme** [SimpleMobileTools/Simple-Contacts#514]
I imported more than 5,000 contacts, to be exact: 5,139. Then I marked all of them and selected them to be deleted. At that point SimpleContacts just closed completely. I tried this multiple times with same result.

**Comment by eklausme** [comment]
I tried to replicate the problem:
1. When I set my screensaver timeout to 10 minutes, application will NOT crash. Though, deleting ca. 5,000 records takes a couple of minutes.
2. When I set my screensaver to 15 seconds, which I usually do, then application terminates or closes completely.

**Comment by tibbi** [comment]
what app version are you using?

**Comment by eklausme** [comment]
I experience this problem with below versions:
1. v5.2.0, free version in Google Play Store
2. v6.9.0 Pro from F-Droid
3. v6.10.2 Pro running in Android-Studio and Emulator

I noticed that even though I selected all records to be deleted, the deletion only works one by one as can be seen either with sqlite3 and constantly checking the number of remaining records or by stepping through deleteContactIds() in class ContactsDao_Impl.

**Comment by tibbi** [comment]
are those contacts stored under the "Phone storage (not visible by other apps)", or where?

**Comment by eklausme** [comment]
Yes, phone storage (not visible by other apps).

**Comment by tibbi** [comment]
right, that is an important detail as it is handled differently than the others, deleting is batched by 50 there. Ill take a look at it, people dont really delete that many contacts often :)
